vnext/Microsoft.ReactNative/Modules/Animated/NativeAnimatedNodeManager.h

```diff
@@ -99,6 +108,9 @@ class NativeAnimatedNodeManager {
   std::unordered_map<std::tuple<int64_t, std::string>, std::vector<std::unique_ptr<EventAnimationDriver>>>
       m_eventDrivers{};
   std::unordered_map<int64_t, std::shared_ptr<AnimationDriver>> m_activeAnimations{};
+  std::unordered_map<int64_t, std::shared_ptr<AnimationDriver>> m_pendingCompletionAnimations{};
+  std::unordered_map<int64_t, std::unordered_set<int64_t>> m_stoppedAnimations{};
+  std::unordered_map<int64_t, std::unordered_set<int64_t>> m_deferredAnimations{};
```

Comment:
There may not be anything to do here, considering the following scenario:
1. Start animation 1 for value X
2. Start animation 2 for value X (will stop animation 1, add animation 2 to pending completion)
3. Start animation 3 for value X (will stop animation 2, add animation 3 to pending completion)

In the current implementation, animation 2 and 3 will be in the pending animation set at the time the completion callback fires for animation 1, and we will attempt to start them simultaneously. Animation 2 will already have been stopped, so there will not be any active animation for it, so it won't get started. Only animation 3 will start, assuming it has not been stopped imperatively.

This could be optimized a bit since we know for certain that animation 2 will be stopped before animation 3 is started. So we never need to add animation 2 to stopped animations (because it never started) and we never need more than 1 animation tag in the deferred animations for a value node tag, because 3 will supercede 2.

However, it may complicate the logic a bit to do this, because when we stop animation 2, we need to check if it is currently the deferred animation before we decide to add it to the stopped animation list.

This also relies on an implementation detail of Animated that there can only be 1 active animation per value node at a time, which isn't likely to change any time soon, but could change in theory.

Reply:
Actually, minor modification, looks like there could be a leak here, because if animation 2 never fires a completion callback, it will never be removed from the stopped animation set. I will go ahead and revise to match the expectations of the Animation API.